### ISSUE TITLE
[BUG]: phalcon4 crypt performance

### ISSUE THREAD

**Issue by davidcorreia** [phalcon/cphalcon#15118]
I have upgraded the project on php7.3 from phalcon 3 to phalcon 4. One thing i have notice, is that the load from the server have increased alot...

one big change i saw was in Phalcon/Crypt

service.php

```php
  $di->setShared('crypt', function () {
    $crypt = new Crypt();
    // Set a global encryption key
    $crypt->setKey('a really long key, 128 chars');

    return $crypt;
  });
```

Example: phalcon3

```php
    $start = microtime(true);
    $crypt = Phalcon\Di::getDefault()->getCrypt();
    $name = $crypt->encryptBase64('teste name');
    for ($i = 0; $i < 1000000; $i++) {
        $crypt->decryptBase64($name);
    }
    echo microtime(true) - $start;
```
2,6897709369659

with phalcon 4 (same code)

26,91615819931

Is this normal behavior?

Thank you

dev & prod environment on php 7.3, tested on cli

**Comment by Jeckerson** [comment]
Fast tested on laptop.

* 128 characters key - `54.802676916122` sec
* 3 characters key - `57.090064048767` sec

Probably problem is not in key length.

**Comment by davidcorreia** [comment]
i think the problem is with $crypt->decryptBase64() and not the key size

**Comment by zsilbi** [comment]
Fixed in #15160

**Comment by zsilbi** [comment]
> i think the problem is with $crypt->decryptBase64() and not the key size

The problem was that the available openssl ciphers were collected and checked before every decrypt.
